### ISSUE TITLE
Build issue when RuntimeIdentifier removed : cross platform package generation from Visual Studio

### ISSUE THREAD

**Issue by jackbond** [microsoft/service-fabric#96]
With the latest SDK, I removed the following from a .netcore actor project file...

`<RuntimeIdentifier>win7-x64</RuntimeIdentifier>`

and I received the following error during build...

`Error	MSB3073	The command "dotnet "obj\\FabActUtilTemp\FabActUtil.dll" /spp:"PackageRoot" /t:manifest /sp:"CoreActor" /in:"bin\Debug\netcoreapp2.0\\CoreActor.dll" /arp:"c:\temp\SFBuildIssue\CoreActor\bin\Debug\netcoreapp2.0\\" " exited with code -2147450745.	CoreActor	C:\Users\jackh\.nuget\packages\microsoft.servicefabric.actors\3.1.262\build\netcoreapp2.0\Microsoft.ServiceFabric.Actors.targets	22	`

I removed the RuntimeIdentifier because it was causing a couple of Nuget packages to complain. It seems strange that the core actor templates include this RuntimeIdentifier by default. Isn't the new SDK and SF packages supposed to support non windows platforms?

**Comment by ashishnegi** [comment]
@jackbond Can you share the steps taken to create the project ? Did you use `dotnet cli` or `Visual Studio` ?

**Comment by ashishnegi** [comment]
@jackbond I created a sample project in VS and used latest `6.2.262` packages. I added Actor code and removed `runtimeidentifier`. And it is building fine.

Can you tell how you created your app ? What is Visual Studio version and SF VS plugin version ?
Please share a sample repro project.

**Comment by jackbond** [comment]
Was it a core project? 

[coreissue.zip](https://github.com/Microsoft/service-fabric/files/1944341/coreissue.zip)

**Comment by jackbond** [comment]
@ashishnegi Do you have a time frame for a fix / updated Nuget? I have dozens of SF projects and I'd rather avoid a temporary workaround. Thanks.

**Comment by ashishnegi** [comment]
@jackbond If you can confirm that this works for you and you face no more errors, we can ship the fix in next release.

cc @amanbha @ravipal for failure in fabactutil in latest packages and fix of adding `fabactutil.runtimeconfig.json`

**Comment by Flern** [comment]
@ashishnegi I had a similar problem and can confirm that the fix worked for me, both by putting the JSON file in `.\obj\FabActUtilTemp` and then in the `%home%\.nuget\packages\microsoft.servicefabric.actors\3.1.269\build\netcoreapp2.0` location.  Both resolved the error message when the RuntimeIdentifier tag was removed.

Note that I am using the 3.1.269 SDK as well as the RC1 version of the .NET Core SDK, so my JSON file contains the following (I still put it in the `netcoreapp2.0` folder in the NuGet package, though):

```
{
  "runtimeOptions": {
    "tfm": "netcoreapp2.1",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "2.1.0-rc1"
    }
  }
}
```

**Comment by ashishnegi** [comment]
@Flern @jackbond  Thanks for trying the fix.
Developing cross platform packages from Visual Studio is untested yet. We are working on it.
To give the best experience, we need to know your requirements better. 
What is your development/test/deployment workflow ? 
Do you deploy on linux or windows ? Do you prefer standalone published packages (no need to install dotnet on prod machine) ?

**Comment by Flern** [comment]
@ashishnegi 
I don't have a lot of definition around my workflow as this was largely an experiment.  I have issues with not being able to install the Service Fabric SDKs in my CI build docker containers, so I was looking at moving my entire SF cluster to Linux for deployment (this first step was building on Linux docker containers, and I don't know how I ran into this issue in the first place).  Since I do deliver to customer sites that include both Windows and Linux clusters, I would like to be able to build and deploy to both with one CI tool chain.  Hosting in the Service Fabric environment, it would also be beneficial to have standalone published packages.  However, I believe the Service Fabric integration is a parallel or different use case / issue.  Sorry I couldn't be more help.

**Comment by luisgoncalves** [comment]
Hello,

Has this been fixed in latest `microsoft.servicefabric.actors` ? Thanks

**Comment by ashishnegi** [comment]
@luisgoncalves This error came when generating cross platform packages from Visual studio.
Is your scenario same ? Are you generating Linux package ?
@amanbha Have we fixed the fabactutil issue ?

**Comment by luisgoncalves** [comment]
@ashishnegi Yrs, I'm generating a Linux package on a Windows machine. For now I used `<UpdateServiceFabricManifestEnabled Condition="'$(RuntimeIdentifier)' != 'win7-x64'">False</UpdateServiceFabricManifestEnabled>` to workaound...

**Comment by anantshankar17** [comment]
This is fixed in the latest 6.5 release (sdk version 3.4.XX).
